crypto/dilithium/ml_dsa.c

```diff
@@ -25,34 +26,34 @@
 // depending on platform support.
 
 int ml_dsa_65_keypair(uint8_t *public_key  /* OUT */,
-                       uint8_t *secret_key /* OUT */) {
+                      uint8_t *secret_key  /* OUT */) {
   ml_dsa_params params;
   ml_dsa_65_params_init(&params);
-  return crypto_sign_keypair(&params, public_key, secret_key);
+  return (crypto_sign_keypair(&params, public_key, secret_key) == 0);
 }
 
-int ml_dsa_65_sign(uint8_t *sig                /* OUT */,
-                    size_t *sig_len            /* OUT */,
-                    const uint8_t *message     /* IN */,
-                    size_t message_len         /* IN */,
-                    const uint8_t *ctx         /* IN */,
-                    size_t ctx_len             /* IN */,
-                    const uint8_t *secret_key  /* IN */) {
+int ml_dsa_65_sign(const uint8_t *secret_key  /* IN */,
+                   uint8_t *sig               /* OUT */,
+                   size_t *sig_len            /* OUT */,
+                   const uint8_t *message     /* IN */,
+                   size_t message_len         /* IN */,
+                   const uint8_t *pre         /* IN */,
+                   size_t pre_len             /* IN */) {
   ml_dsa_params params;
   ml_dsa_65_params_init(&params);
   return crypto_sign_signature(&params, sig, sig_len, message, message_len,
```

Comment:
how come here and for verify you don't have to do `return verify() == 0`?

Reply:
The reason is that we used to perform this check within the PKEY sign/verify, rather than at the MLDSA level.

In [7435f9b](https://github.com/aws/aws-lc/pull/1963/commits/7435f9b2920753efc71a6461c55be213dc24da65) we replace:
- `ml_dsa_65_sign(sig, siglen, tbs, tbslen, NULL, 0, key->priv) != 0`
- `ml_dsa_65_verify(tbs, tbslen, sig, siglen, NULL, 0, key->pub) != 0`

with:
- `!pqdsa->method->sign(key->secret_key, sig, sig_len, message, message_len, NULL, 0)`
- `!pqdsa->method->verify(key->public_key, sig, sig_len, message, message_len, NULL, 0)`

at:
-  Sign: https://github.com/jakemas/aws-lc/blob/7435f9b2920753efc71a6461c55be213dc24da65/crypto/dilithium/p_pqdsa.c#L100
- Verify: https://github.com/jakemas/aws-lc/blob/7435f9b2920753efc71a6461c55be213dc24da65/crypto/dilithium/p_pqdsa.c#L132